### PR TITLE
feat: implement CONCAT_WS function for E6 dialect

### PR DESCRIPTION
**Pull request by tkaunlaky-e6**
## Summary
• Implemented CONCAT_WS function for E6 dialect with full Databricks compatibility
• Added comprehensive test coverage for all documented behaviors

## Implementation Details
• **NULL handling**: NULL separator returns NULL, NULL arguments are filtered out
• **Array support**: Arrays are flattened and NULL elements within arrays are removed  
• **Edge cases**: Returns empty string when only separator provided or all args are NULL
• **Performance**: Uses efficient AST manipulation instead of complex CASE statements

## Test Coverage
Added tests covering all Databricks examples:
• `concat_ws(' ', 'Spark', 'SQL')` → `'Spark SQL'`
• `concat_ws('s')` → `''`
• `concat_ws(',', 'Spark', array('S', 'Q', NULL, 'L'), NULL)` → `'Spark,S,Q,L'`
• Multiple additional edge cases

## Test Results
✅ All 857 tests pass - no regressions introduced

### CODE REVIEW

File: sqlglot/dialects/e6.py [hunk 1]
```diff
@@ -1978,6 +1978,57 @@ def string_agg_sql(self: E6.Generator, expression: exp.GroupConcat) -> str:
             # Generate SQL using STRING_AGG/LISTAGG, with separator or default ''
             return self.func("LISTAGG", expr_1, separator or exp.Literal.string(""))
 
+        def concat_ws_sql(self: E6.Generator, expression: exp.ConcatWs) -> str:
```
Comment (Adithyak-0926) [inline sqlglot/dialects/e6.py:1981]: @tkaunlaky-e6 did u also check the databricks parsing logic? we want to achieve name concat_ws to be preserved because it also started supported right?

Comment (tkaunlaky-e6) [reply sqlglot/dialects/e6.py:1981]: @Adithyak-0926 Databricks AST shows that is mapped to Concatws.

File: sqlglot/dialects/e6.py [hunk 2]
```diff
@@ -2018,6 +2018,57 @@ def string_agg_sql(self: E6.Generator, expression: exp.GroupConcat) -> str:
...
+                return "''"
+            
+            if len(all_elements) == 1:
+                # Single element case
+                return all_elements[0]
+            
+            # Multiple elements: create array and join with separator
+            # Build: ARRAY_TO_STRING(ARRAY[element1, element2, ...], separator)
+            elements_list = ", ".join(all_elements)
+            return f"ARRAY_TO_STRING(ARRAY[{elements_list}], {self.sql(separator)})"
```
Comment (gauravdawar-e6) [inline sqlglot/dialects/e6.py:2070]: instead of hard coding "ARRAY", can't we use the node or expression?

**Review by ranjanankur314 — DISMISSED** [review]
Looks good

**Review by shreyas2201 — APPROVED** [review]

**Review by aravindh-e6x — DISMISSED** [review]

**Comment by tkaunlaky-e6** [comment]
@ranjanankur314 @gauravdawar-e6 @aravindh-e6x 
I rebased on latest main to clean up the commit history - same CONCAT_WS code, just without the merge commits now.
I think GitHub dismissed the approvals because of the rebase.

**Review by ranjanankur314 — APPROVED** [review]
